lightning/src/ln/channelmanager.rs

```diff
@@ -987,6 +987,10 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref> ChannelMan
 
 				// OUR PAYMENT!
 				// final_expiry_too_soon
+				// Final node can't rely on checking a CLTV_EXPIRY_DELTA which enforces by its own length CLTV_CLAIM_BUFFER so make sure
+				// we don't accept incoming HTLC we wouldn't have time to claim with a worst-case broadcast scenario
+				// Also, in case of unknown payment hash, makes sure payment logic has time to fail backward HTLC before our onchain
+				// logic trigger a channel closure.
```

Comment:
Grammar/phrasing nit:

rephrase to: "Also: ensure that, in the case of an unknown payment hash, our payment logic has enough time to fail the HTLC backward before our onchain logic triggers a channel closure."